### PR TITLE
Data validation for forms

### PR DESCRIPTION
**Pull request by suratdas**
Login and registration button is enabled only when valid data is entered.

### CODE REVIEW

File: src/components/TestDetailsModal.tsx [hunk 1]
```diff
@@ -152,6 +152,7 @@ const TestDetailsModal: React.FunctionComponent<{
   );
 
   useHotkeys("d", () => setIsDiffShown((isDiffShown) => !isDiffShown));
+  useHotkeys("ESC", () => handleClose());
```
Comment (pashidlos) [inline src/components/TestDetailsModal.tsx:155]: could you make separate PR for this?

File: src/pages/ProjectPage.tsx [hunk 2]
```diff
@@ -112,7 +112,7 @@ const ProjectPage = () => {
     setFilteredTestRuns(
       testRuns.filter(
         (t) =>
-          t.name.includes(query) && // by query
+          t.name.toLowerCase().includes(query.toLowerCase()) && // by query
```
Comment (pashidlos) [inline src/pages/ProjectPage.tsx:115]: could you make separate PR for this?
I'll need to create separate ticket to have it in release log

**Review by pashidlos — CHANGES_REQUESTED** [review]
created a ticket for it with some description 
would like to add validation in the same manner across all form 
https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/172

File: src/pages/ProjectPage.tsx (hunk 2 above)
Comment (pashidlos) [reply src/pages/ProjectPage.tsx:115]: https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/173

File: src/components/TestDetailsModal.tsx (hunk 1 above)
Comment (pashidlos) [reply src/components/TestDetailsModal.tsx:155]: https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/174

**Comment by pashidlos** [comment]
@shankybnl thank you for contribution!
I invited you in organisation with write access to frontend/backend repo 
create branches from this repo instead your fork, this will fix cypress tests

**Comment by suratdas** [comment]
Split the pull requests. I still kept this pull request from my form open for the form controls. I will keep adding changes as new branch instead of fork.
https://github.com/Visual-Regression-Tracker/Visual-Regression-Tracker/issues/172 -This is partially doe.
@pashidlos , I have not created a twitter account yet.

**Comment by shankybnl** [comment]
> @shankybnl thank you for contribution!
> I invited you in organisation with write access to frontend/backend repo 
> create branches from this repo instead your fork, this will fix cypress tests

Thank you @pashidlos for giving the access.

**Comment by suratdas** [comment]
Screen Shot 2021-01-15 at 12 40 01 AM
Screen Shot 2021-01-15 at 12 52 05 AM
These are the types of validation that are part of this pull request.

**Review by pashidlos — CHANGES_REQUESTED** [review]
could you check this [react-material-ui-form-validator](https://www.npmjs.com/package/react-material-ui-form-validator) as it may simplify the job ?

File: src/components/BaseModal.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ interface IProps {
   title: string;
   content: React.ReactNode;
   submitButtonText: string;
+  isDisabled :boolean;
```
[the remark this answers is not in the document]
Comment (suratdas) [reply src/components/BaseModal.tsx]: Fixed and used react-material-ui-form-validator

File: src/components/RegisterForm.tsx [hunk 4]
```diff
@@ -23,21 +23,34 @@ const RegisterForm = () => {
...
+        login(dispatch, email, password);
+      })
       .catch((err) =>
         enqueueSnackbar(err, {
           variant: "error",
         })
       );
   };
 
+  ValidatorForm.addValidationRule(
```
Comment (pashidlos) [inline src/components/RegisterForm.tsx:39]: could this be reused not to duplicate in every file?

**Comment by pashidlos** [comment]
@suratdas it looks awesome!

File: src/components/RegisterForm.tsx (hunk 4 above)
Comment (pashidlos) [reply src/components/RegisterForm.tsx:39]: could this be replaced with `minStringLength` rule? 
I found list of implemented rules from here
https://www.npmjs.com/package/react-form-validator-core

Comment (suratdas) [reply src/components/RegisterForm.tsx]: Done

**Review by pashidlos — APPROVED** [review]
